src/Service/SoftDelete.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\User;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class SoftDelete {
```

Comment:
I think UserSoftDelete class with $user mandatory in constructor could be more natural.